### PR TITLE
ref: Rust 1.78.0 lints

### PR DESCRIPTION
**Pull request by jjbayer**
#skip-changelog

### CODE REVIEW

**Review by Dav1dde — APPROVED** [review]

File: relay-cardinality/src/redis/limiter.rs [hunk 1]
```diff
@@ -920,7 +920,7 @@ mod tests {
                         cardinality: 1
                     }]
                 );
-                assert!(rejected.reports.get(&limits[3]).is_none());
+                assert!(!rejected.reports.contains_key(&limits[3]));
```
Comment (iker-barriocanal) [inline relay-cardinality/src/redis/limiter.rs:923]: Clippy... we need to talk.

File: relay-server/src/services/store.rs [hunk 2]
```diff
@@ -1278,25 +1278,6 @@ struct UserReportKafkaMessage {
     event_id: EventId,
 }
 
-#[derive(Clone, Debug, Serialize)]
-struct SessionKafkaMessage {
```
Comment (iker-barriocanal) [inline relay-server/src/services/store.rs:1282]: Was this unused?

**Review by iker-barriocanal — APPROVED** [review]

**Comment by Dav1dde** [comment]
@jjbayer please fix the title/commit message

File: relay-server/src/services/store.rs (hunk 2 above)
Comment (jjbayer) [reply relay-server/src/services/store.rs:1282]: Yeah, not sure why it wasn't caught before.

**Review by olksdr — APPROVED** [review]

**Pull request merged**